libsoftwarecontainer/src/gateway/waylandgateway.cpp

```diff
@@ -40,6 +41,11 @@ WaylandGateway::~WaylandGateway()
 
 ReturnCode WaylandGateway::readConfigElement(const json_t *element)
 {
+    if (m_enabled == true) {
+        log_debug() << "WaylanGateway is already enabled";
+        return ReturnCode::SUCCESS;
```

Comment:
This is imho not the desired behaviour. With this, one can supply bad configurations once there is one valid configuration sent with enabled to true. We should ALWAYS read the config to make sure it is well-formed. In this case it makes sense to read it into a temporary var first which one can then or with `m_enabled`.